pkg/types/kustomization.go

```diff
@@ -79,6 +81,10 @@ type Kustomization struct {
 
 	// ImageTags is a list of ImageTag for changing image tags
 	ImageTags []ImageTag `json:"imageTags,omitempty" yaml:"imageTags,omitempty"`
+
+	// JSONPatches is a list of JSONPatch for applying JSON patch.
+	// The JSON patch is documented at http://jsonpatch.com/.
+	JSONPatches []JSONPatch `json:"jsonPatches,omitempty" yaml:"jsonPatches,omitempty"`
```

Comment:
> So is this it? Any more kinds of patching besides SMP and 6902? @mengqiy ?

Re. @monopole 's earlier comment. There is also JSON merge patch (https://tools.ietf.org/html/rfc7386) which is very similar to SMP. The only difference is not supporting merging list.

Are you going to add another field to support that?
@monopole @Liujingfang1 WDYT?

Reply:
It's also easy to imaging someone asking for (Larry Wall) style patch, which could look like

diffPatches
 - resFile: {some resource file}
   patchFile: {some diff style patch}

Maybe this needs a small design doc so we can work it out.

Reply:
@mengqiy,  @monopole and I also discussed another possible format for patches with a `kind` field.
```
patches:
- path: some patch
  kind: JSON  # default is stategic merge patch;
  target:              # This field is required only for JSON patch
      kind: Service
      name: some service    
``` 
This format can support specifying different kinds of patches and can be easily extended to support `diffPatches`.

The only drawback of this format is that it's not compatible with our current `patches` format. I'm currently experimenting if we can achieve this format with backward compatibility. If not, I will write a design doc for different approaches and we can compare.

Reply:
The format @Liujingfang1 described above was something in my mind. I believe that format is more extensible for new patch types. But it may break back compatibility :/